File: Legacy/Objective-C/Neocom/Neocom/NCTintImageView.m
//
//  NCTintImageView.m
//  Neocom
//
//  Created by Artem Shimanski on 14.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "NCTintImageView.h"

@implementation NCTintImageView

- (void) awakeFromNib {
	[super awakeFromNib];
	self.layer.masksToBounds = YES;
	//Storyboards fix template image
	UIColor* tintColor = self.tintColor;
	self.tintColor = nil;
	self.tintColor = tintColor;
}


#if TARGET_INTERFACE_BUILDER
- (void) prepareForInterfaceBuilder {
	self.layer.masksToBounds = YES;
	//if (self.image.renderingMode == UIImageRenderingModeAlwaysTemplate)
	self.image = [self.image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
}
#endif

@end

File: Neocom/Neocom/NCTintImageView.swift
//
//  NCTintImageView.swift
//  Neocom
//
//  Created by Artem Shimanski on 14.12.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import Foundation

@IBDesignable
class NCTintImageView: NCImageView {
	override func awakeFromNib() {
		let color = tintColor
		tintColor = nil
		tintColor = color
	}
	
	override var intrinsicContentSize: CGSize {
		return image != nil ? super.intrinsicContentSize : CGSize.zero
	}
	
#if TARGET_INTERFACE_BUILDER
	override func prepareForInterfaceBuilder() {
		image = image?.withRenderingMode(.alwaysTemplate)
	}
#endif
}
